crates/rpc/rpc-types-compat/src/proof.rs

```diff
@@ -23,6 +25,11 @@ pub fn from_primitive_account_proof(proof: AccountProof) -> EIP1186AccountProofR
         nonce: info.nonce,
         storage_hash: proof.storage_root,
         account_proof: proof.proof,
-        storage_proof: proof.storage_proofs.into_iter().map(from_primitive_storage_proof).collect(),
+        storage_proof: proof
+            .storage_proofs
+            .into_iter()
+            .zip(slots)
+            .map(|(proof, slot)| from_primitive_storage_proof(proof, slot))
```

Comment:
@rkrasiuk is it OK to assume same order if input slots and outputted proofs here?

Reply:
it is the case rn, but i'd rather avoid such assumption since it's not documented anywhere, not covered in tests and there is no such expectation atm to proof generation